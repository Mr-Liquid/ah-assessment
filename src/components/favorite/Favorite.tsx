import { FavoriteIcon } from '../icons';

type FavoriteProps = {
  id: string;
  favorite: boolean;
  onFavorite: (data: { id: string; actionType: 'add' | 'remove' }) => void;
};

export const Favorite = ({ id, favorite, onFavorite }: FavoriteProps) => {
  const handleFavoriteClick = () => {
    onFavorite({
      id,
      actionType: favorite ? 'remove' : 'add',
    });
  };

  return (
    <button onClick={handleFavoriteClick} aria-pressed={favorite}>
      <FavoriteIcon favorite={favorite} />
    </button>
  );
};

export default Favorite;
