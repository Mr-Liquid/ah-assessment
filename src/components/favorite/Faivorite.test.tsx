import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Favorite } from './Favorite';

describe('Favorite Component', () => {
  it('matches snapshot favorite is set', () => {
    const { container } = render(
      <Favorite favorite={true} id="testId" onFavorite={() => {}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot favorite is unset', () => {
    const { container } = render(
      <Favorite favorite={false} id="testId" onFavorite={() => {}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
