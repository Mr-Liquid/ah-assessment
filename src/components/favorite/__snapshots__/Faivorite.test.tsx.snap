// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Favorite Component > matches snapshot favorite is set 1`] = `
<button
  aria-pressed="true"
>
  <svg
    class="cursor-pointer fill-gray-400 "
    fill="none"
    height="21"
    stroke="#8899a4"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="21"
  >
    <path
      d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
    />
  </svg>
</button>
`;

exports[`Favorite Component > matches snapshot favorite is unset 1`] = `
<button
  aria-pressed="false"
>
  <svg
    class="cursor-pointer  "
    fill="none"
    height="21"
    stroke="#8899a4"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="21"
  >
    <path
      d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
    />
  </svg>
</button>
`;
